OSDOCS-4012: Add platform Operator and RukPak docs (#52907)
openshift/openshift-docs: modules/olm-installing-po-after.adoc

// Module included in the following assemblies:
//
// * operators/admin/olm-managing-po.adoc

:_content-type: PROCEDURE
[id="olm-installing-po-after_{context}"]
= Installing platform Operators after cluster creation

> As a cluster administrator, you can install platform Operators after cluster creation on clusters that have enabled the `TechPreviewNoUpgrades` feature set by using the cluster-wide `PlatformOperator` API.

Reviewer: Maybe we can add the below command for users.
Note that, once enabled, tech preview features may not be disabled.
```console
MacBook-Pro:~ jianzhang$ oc patch featuregate cluster -p '{"spec": {"featureSet": "TechPreviewNoUpgrade"}}' --type=merge
featuregate.config.openshift.io/cluster patched
MacBook-Pro:~ jianzhang$ oc get pods -n openshift-platform-operators
NAME                                                     READY   STATUS    RESTARTS   AGE
platform-operators-controller-manager-77d5d88cc6-w48jj   2/2     Running   0          2m4s
platform-operators-rukpak-core-c5978759-pxvq9            2/2     Running   0          2m4s
platform-operators-rukpak-webhooks-758764ff88-855d6      1/1     Running   0          2m4s
platform-operators-rukpak-webhooks-758764ff88-wh42m      1/1     Running   0          2m4s

```

// Module included in the following assemblies:
//
// * operators/admin/olm-managing-po.adoc

:_content-type: PROCEDURE
[id="olm-installing-po-after_{context}"]
= Installing platform Operators after cluster creation

As a cluster administrator, you can install platform Operators after cluster creation on clusters that have enabled the `TechPreviewNoUpgrades` feature set by using the cluster-wide `PlatformOperator` API.

.Prerequisites

- Access to an {product-title} cluster using an account with `cluster-admin` permissions.
- Enable the `TechPreviewNoUpgrades` feature set on the cluster. See _Additional resources_ for details.

> - Install the `oc` command to your local system.

Reviewer 2: Is there a way to make these parallel in construction? (might not be possible without being too convoluted)

Author: Was trying to match the styling from another assembly but, this has clearly gotten awkward so I will abandon that and have now made them parallel (also moved them up to the assembly-level since they were just being repeated).

.Procedure

. Choose a platform Operator from the supported set of OLM-based Operators. See _Additional resources_ for the list of this set and details on current limitations.

> . Create an `PlatformOperator` object YAML file for your chosen platform Operator, for example a `cert-manager.yaml` file for the {cert-manager-operator}:

Reviewer 2: ```suggestion
. Create a `PlatformOperator` object YAML file for your chosen platform Operator, for example a `cert-manager.yaml` file for the {cert-manager-operator}:
```

+
[source,yaml]
----
apiVersion: platform.openshift.io/v1alpha1
kind: PlatformOperator
metadata:
  name: cert-manager
spec:
  package:
    name: openshift-cert-manager-operator
----

> . Create the `PlatformOperator` object:

Reviewer 2: ```suggestion
. Create the `PlatformOperator` object by running the following command:
```

+
[source,terminal]
----
$ oc apply -f cert-manager.yaml
----

> . Wait until the `cert-manager` platform Operator has reported an `Installed=true` status condition:

Reviewer 2: Might want to say what the command is doing, e.g.
```suggestion
. Check the status of the `cert-manager` platform Operator by running the following command:
```

+
[source,terminal]
----
$ oc get platformoperator cert-manager -o yaml
----
+
.Example output
[source,yaml]
----
...
status:
  activeBundleDeployment:
    name: cert-manager
  conditions:
  - lastTransitionTime: "2022-10-24T17:24:40Z"
    message: Successfully applied the cert-manager BundleDeployment resource
    reason: InstallSuccessful
    status: "True"

>     type: Installed

Reviewer 2: Then you could add callouts to the parts they are looking for explaining what success looks like, or possibly just add it as a result line under this block.

Just a thought!
